### ISSUE TITLE
Issue with parsing string as single type.

### ISSUE THREAD

**Issue by colton-boi** [SkriptLang/Skript#5262]
### Skript/Server Version

```markdown
[02:42:22 INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[02:42:22 INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[02:42:22 INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[02:42:22 INFO]: [Skript] Server Version: git-Pufferfish-52 (MC: 1.19.3)
[02:42:22 INFO]: [Skript] Skript Version: 2.6.3
[02:42:22 INFO]: [Skript] Installed Skript Addons: 
[02:42:22 INFO]: [Skript]  - skript-gui v1.3-beta2 (https://github.com/APickledWalrus/skript-gui)
[02:42:22 INFO]: [Skript]  - DiSky v4.4.3
[02:42:22 INFO]: [Skript]  - skript-reflect v2.3 (https://github.com/TPGamesNL/skript-reflect)
[02:42:22 INFO]: [Skript]  - skRayFall v1.9.28 (https://sk.rayfall.net/)
[02:42:22 INFO]: [Skript]  - SkQuery v4.1.7
[02:42:22 INFO]: [Skript]  - SkriptJSON v1.0.0
[02:42:22 INFO]: [Skript]  - SkBee v2.5.1 (https://github.com/ShaneBeee/SkBee)
[02:42:22 INFO]: [Skript]  - FunkySk v1.0.2
[02:42:22 INFO]: [Skript]  - skript-holo v1.0.2 (https://github.com/Blueyescat/skript-holo)
[02:42:22 INFO]: [Skript] Installed dependencies: 
[02:42:22 INFO]: [Skript]  - Vault v1.7.3-b131

yes I know I have a lot of addons, shush
```

### Bug Description

Setting a variable to `string parsed as "test%string%"` results in the error `variable can only be set to one object, not more`, while only a singular object is returned. ![image](https://user-images.githubusercontent.com/71715967/209260659-5354fb8a-57db-4275-8f09-0e8e6c4feae3.png)

### Expected Behavior

No error and the variable being set

### Steps to Reproduce

All in the bug description. One way around this is to, instead of setting it to `string parsed as "test%string%"` setting it to `second element of (string parsed as "%string%est%string%")` but that shouldn't be necessary

You are also unable to set the variable to `first element of string parsed as "test%string%"`
![image](https://user-images.githubusercontent.com/71715967/209260883-75ad2298-17b6-40c6-8e0c-3f9e22a51130.png)

### Agreement

- [x] I have read the guidelines above and affirm I am following them with this report.

**Comment by TheLimeGlass** [comment]
This has always been intended behavior. Skript cannot grab the string value at parse time, only at runtime. This means Skript can't tell if you used single or multiple expression types in your string.

So in order to be able to support multiple expression types in the string `%text% tier %number%` the expression has to return false indicating multiple values can be present when the custom text parse is used.

A solution could be making the parsed as custom text a literal String which we've been experimenting with in other things, or set it to a variable and use the soon to be implemented type hints of variables, but that's not something for awhile.

**Comment by TheLimeGlass** [comment]
Closing soon after other team members have seen.

**Comment by colton-boi** [comment]
If that's the case, just want to comment that running `!set {d} to first element of (uncolored name of tool parsed as "text%string%")` gives the following error
`[04:16:05 ERROR]: #!#! 
[04:16:05 ERROR]: #!#! [Skript] Severe Error:
[04:16:05 ERROR]: #!#! 
[04:16:05 ERROR]: #!#! You're running outdated version of Skript! Please try updating it NOW; it might fix this.
[04:16:05 ERROR]: #!#! Run /sk update check to get a download link to latest Skript!
[04:16:05 ERROR]: #!#! You will be given instructions how to report this error if it persists after update.
[04:16:05 ERROR]: #!#! 
[04:16:05 ERROR]: #!#! Stack trace:
[04:16:05 ERROR]: #!#! java.lang.ArrayStoreException: java.lang.String
[04:16:05 ERROR]: #!#!     at Skript (9).jar//ch.njol.skript.expressions.ExprElement.get(ExprElement.java:87)
[04:16:05 ERROR]: #!#!     at Skript (9).jar//ch.njol.skript.lang.util.SimpleExpression.getArray(SimpleExpression.java:99)
[04:16:05 ERROR]: #!#!     at Skript (9).jar//ch.njol.skript.effects.EffChange.execute(EffChange.java:274)
[04:16:05 ERROR]: #!#!     at Skript (9).jar//ch.njol.skript.lang.Effect.run(Effect.java:50)
[04:16:05 ERROR]: #!#!     at Skript (9).jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:61)
[04:16:05 ERROR]: #!#!     at Skript (9).jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:89)
[04:16:05 ERROR]: #!#!     at Skript (9).jar//ch.njol.skript.command.Commands.handleEffectCommand(Commands.java:290)
[04:16:05 ERROR]: #!#!     at Skript (9).jar//ch.njol.skript.command.Commands$3$1.call(Commands.java:220)
[04:16:05 ERROR]: #!#!     at Skript (9).jar//ch.njol.skript.command.Commands$3$1.call(Commands.java:217)
[04:16:05 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftFuture.run(CraftFuture.java:88)
[04:16:05 ERROR]: #!#!     at org.bukkit.craftbukkit.v1_19_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483)
[04:16:05 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1475)
[04:16:05 ERROR]: #!#!     at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:443)
[04:16:05 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1399)
[04:16:05 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1175)
[04:16:05 ERROR]: #!#!     at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:318)
[04:16:05 ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:833)
[04:16:05 ERROR]: #!#! 
[04:16:05 ERROR]: #!#! Version Information:
[04:16:05 ERROR]: #!#!   Skript: 2.6.3 (OUTDATED)
[04:16:05 ERROR]: #!#!     Flavor: skriptlang-github
[04:16:05 ERROR]: #!#!     Date: 00:55:22.847925631
[04:16:05 ERROR]: #!#!   Bukkit: 1.19.3-R0.1-SNAPSHOT
[04:16:05 ERROR]: #!#!   Minecraft: 1.19.3
[04:16:05 ERROR]: #!#!   Java: 17.0.2 (OpenJDK 64-Bit Server VM 17.0.2+8-86)
[04:16:05 ERROR]: #!#!   OS: Linux amd64 4.15.0-200-generic
[04:16:05 ERROR]: #!#! 
[04:16:05 ERROR]: #!#! Server platform: Paper
[04:16:05 ERROR]: #!#! 
[04:16:05 ERROR]: #!#! Current node: null
[04:16:05 ERROR]: #!#! Current item: set {d} (as java.lang.Object) to the first element of uncoloured the name of  tool of event-living entity parsed as VALID %string%
[04:16:05 ERROR]: #!#! 
[04:16:05 ERROR]: #!#! Thread: Server thread
[04:16:05 ERROR]: #!#! 
[04:16:05 ERROR]: #!#! Language: english
[04:16:05 ERROR]: #!#! Link parse mode: DISABLED
[04:16:05 ERROR]: #!#! 
[04:16:05 ERROR]: #!#! End of Error.
[04:16:05 ERROR]: #!#!`

**Comment by TheLimeGlass** [comment]
> If that's the case, just want to comment that running `!set {d} to first element of (uncolored name of tool parsed as "text%string%")` gives the following error https://pastebin.com/CD6p2iFR

I cannot reproduce with a diamond sword named "testExample"

**Comment by TPGamesNL** [comment]
The pattern of ExprParse is already a literal, so analyzing the pattern to determine whether it will always be single during init is possible and won't break anything. It could also be analyzed to better specify the return type of ExprParse.

The ArrayStoreException will be fixed in #5274.
